Content.Shared/Construction/Conditions/NoUnstackableInTile.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Content.Shared.Tag;
+using JetBrains.Annotations;
+using Robust.Shared.Map;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map.Components;
+
+namespace Content.Shared.Construction.Conditions
+{
+    [UsedImplicitly]
+    [DataDefinition]
+    public sealed class NoUnstackableInTile : IConstructionCondition
+    {
+        public const string GuidebookString = "construction-step-condition-no-unstackable-in-tile";
+        public bool Condition(EntityUid user, EntityCoordinates location, Direction direction)
+        {
+            var tagSystem = IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<TagSystem>();
+
+            if (AnyUnstackableTiles(location, tagSystem))
+                return false;
+
+            return true;
+        }
+
+        public static bool AnyUnstackableTiles(EntityCoordinates location, TagSystem tagSystem)
+        {
+            var entityManager = IoCManager.Resolve<IEntityManager>();
+
+            var gridUid = location.GetGridUid(entityManager);
+            if (gridUid == null)
+                return false;
+
+            if (!entityManager.TryGetComponent<MapGridComponent>((EntityUid)gridUid, out var grid))
```

Comment:
This cast looks very sussy?

Reply:
I don't know how to discard the ? from `EntityUid?` without it. I really dislike the cast too.

Reply:
.Value